File: bit_manipulation/and_pair_xor_sum/and_pair_xor_sum.py

def count_and_pair_xor_sum(nums_1: list[int], nums_2: list[int]) -> int:  # LeetCode Q.1835.
    # Key: (a & b) ^ (a & c) = a & (b ^ c).
    xor_value_1, xor_value_2 = 0, 0
    for num in nums_1:
        xor_value_1 ^= num

    for num in nums_2:
        xor_value_2 ^= num

    return xor_value_1 & xor_value_2

File: bit_manipulation/and_pair_xor_sum/and_pair_xor_sum.cpp
#include <vector>
using namespace std;

int count_and_pair_xor_sum(vector<int> &nums_1, vector<int> &nums_2)
{ // LeetCode Q.1835.
    // Key: (a & b) ^ (a & c) = a & (b ^ c).
    int xor_value_1 = 0;
    for (auto num : nums_1)
    {
        xor_value_1 ^= num;
    }
    int xor_value_2 = 0;
    for (auto num : nums_2)
    {
        xor_value_2 ^= num;
    }
    return xor_value_1 & xor_value_2;
}
